mods/default/chests.lua

```diff
@@ -20,6 +20,9 @@ function default.chest.chest_lid_obstructed(pos)
 	local above = {x = pos.x, y = pos.y + 1, z = pos.z}
 	local def = minetest.registered_nodes[minetest.get_node(above).name]
 	-- allow ladders, signs, wallmounted things and torches to not obstruct
+	if def.drawtype == "mesh" then
+		minetest.remove_node(above)
```

Comment:
You seem to be deleting arbitrary mesh nodes here. This should be more specific; in it's current form it's prone to abuse and mod breakage both within MTG and outside of MTG (concerning 3rd-party mods).